ouroboros-network/test/Test/Ouroboros/Network/PeerSelection.hs

```diff
@@ -1915,6 +1917,13 @@ _governorFindingPublicRoots targetNumberOfRootPeers readDomains =
     tracer :: Show a => Tracer IO a
     tracer  = Tracer (BS.putStrLn . BS.pack . show)
 
+    dnsActions :: DNSActions ResolvConf DNS.Resolver IOException IO
+    dnsActions = DNSActions {
+            dnsResolverResource = resolverResource,
+            dnsAsyncResolverResource = asyncResolverResource,
+            dnsLookupAWithTTL = lookupAWithTTL
+    }
```

Comment:
and here.

Reply:
Here regards `IOSim` and it is to build `mockDNSActions`

Reply:
but it is still in `IO` monad: `dnsActions :: DNSActions ResolvConf DNS.Resolver IOException IO` and it's exactly the same as `ioDNSActions` (modulo the missing windows only member).